tests/main/prepare-image-validation-sets/task.yaml

```diff
@@ -0,0 +1,162 @@
+    gendeveloper1 sign-model < ./asserts/core-20.json > model.assert
+
+    echo Expose the needed assertions through the fakestore
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+    cp foo.assert "$STORE_DIR/asserts"
+    cp bar.assert "$STORE_DIR/asserts"
+
+    # It is not enough to copy the assertions, we must also ack them otherwise we
+    # will get an error about not being able to resolve the account key
+    snap ack "$STORE_DIR/asserts/testrootorg-store.account-key"
+    snap ack "$STORE_DIR/asserts/developer1.account"
+    snap ack "$STORE_DIR/asserts/developer1.account-key"
+    
+    # We now add all the required snaps to the fake store. It hardly matter which revision
+    # we give to them, as the fake store does not handle requests of specific revisions. Currently
+    # the fake-store will just return whatever revision there is.
+    echo "Adding snaps to the fake store"
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls core_*.snap)" 99T7MUlRhtI3U0QFgl5mXXESAiSwt776
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls pc_*.snap)" UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls pc-kernel_*.snap)" pYVQrBcKmBa0mZ4CCN7ExT6jH8rY1hza
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls core20_*.snap)" DLqre5XGLbDqg9jPtiAhRRjDuPVa5X1q
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls snapd_*.snap)" PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4
+    install_snap_to_fakestore test-snapd-sh 13
+
+    # Use the fake-store to verify behaviour, so we can use our own assertions
+    export SNAPPY_FORCE_API_URL=http://$STORE_ADDR
+    echo Running prepare-image
+    snap prepare-image --channel edge --write-revisions --snap core --snap test-snapd-sh ./model.assert $ROOT
+    
+    # The generated seed.manifest will be in working directory.
+    echo Verifying the generated manifest has correct contents
+    MATCH "developer1/bar\s1+$" < seed.manifest
+    MATCH "developer1/foo\s1+$" < seed.manifest
+    MATCH "core\s1+$" < seed.manifest
+    MATCH "core20\s1+$" < seed.manifest
+    MATCH "snapd\s1+$" < seed.manifest
+    
+    echo Verifying the expected revisions were retrieved
+    test -e "$IMAGE/snaps/core20_1.snap"
+    test -e "$IMAGE/snaps/pc_1.snap"
+    test -e "$IMAGE/snaps/pc-kernel_1.snap"
+    test -e "$IMAGE/snaps/snapd_1.snap"
+
+    CURDATE=$(date +"%Y%m%d")
+    test -e "$IMAGE/systems/$CURDATE/snaps/core_1.snap"
+    test -e "$IMAGE/systems/$CURDATE/snaps/test-snapd-sh_13.snap"
+    
+    echo Verifying that the store acknowledged we requested the expected revisions set by validation sets
+    # vs1.json
+    # pc:        1
+    # pc-kernel: 1
+    journalctl -u fakestore | grep 'requested snap "pc" revision 1'
```

Comment:
Use journal-state get-log -u fakestore
This is needed to check the logs for the current test, otherwise it is checking logs for the whole execution

Reply:
Same comment for the other journalctl uses

Reply:
Nice suggestions, thanks!